qiskit/primitives/backend_sampler.py

```diff
@@ -0,0 +1,252 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Sampler implementation for an artibtrary Backend object."""
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+import copy
+from typing import cast, Any
+
+from qiskit.exceptions import QiskitError
+from qiskit.providers.options import Options
+from qiskit.result import QuasiDistribution
+from .base_sampler import BaseSampler
+from .sampler_result import SamplerResult
+from .primitive_job import PrimitiveJob
+
+
+class BackendSampler(BaseSampler):
+    """A :class:`~.BaseSampler` implementation that provides an interface for leveraging
+    the sampler interface from any backend.
+
+    This class provides a sampler interface from any backend and doesn't
+    any measurement mitigation, it just computes the probability distribution
+    from the counts.
+    """
+
+    def __init__(
+        self,
+        backend: "Backend" = None,
```

Comment:
The base `Backend` class doesn't have any methods, including `run`, so this probably isn't well typed.  Perhaps the initialiser should reject backends outside its known version support, in order to narrow the type of `self._backend` internally to `BackendV1 | BackendV2`, which can satisfy everything used in the implementation (assuming of course that both do work).

How come a default argument of `None` is supplied if that's not actually a valid value?

Reply:
This was a weird case, there is a `__new__` with fixed positional arguments defined. I could get it to initalize an object cleanly with backend as the first positional arg. So this was my quick compromise to force it to pass it by name. I guess I could have done:

```suggestion
        *backend: "Backend",
```

instead